core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2076,7 +2076,7 @@ public void delete(AlluxioURI path, DeleteContext context)
       return;
     }
     Metrics.DELETE_PATHS_OPS.inc();
-    try (RpcContext rpcContext = createRpcContext(context);
+    try (RpcContext rpcContext = createRpcContextAsyncJournal(context, journalContext);
         FileSystemMasterAuditContext auditContext =
```

Comment:
if delete non-alluxioonly, will syncMetadata force flush?

Reply:
I think the answer is no.

If you check all callers of `flush()`, we flush when:
1. At the end of an RPC, when the RpcContext is closed (that closes the JournalContext)
2. When MetadataSyncMergeJournalContext is enabled, manually flush but only flushes to the AsyncJournalWriter so it's still async.
https://github.com/Alluxio/alluxio/blob/cb8d7cabd7ea83b9561e234bd93dc44420dd7895/core/server/common/src/main/java/alluxio/master/journal/JournalContext.java#L37

Reply:
Honestly the above are not an explicitly written protocol. I spent some time digging the code and I'm only 90% sure I'm right. For that 10% I can only say I'll keep an eye and fix those extra `flush()` if I later find any. We now have a bit too many layers of wrappers so it's hard to be sure the behavior doesn't change at any layer.. I think later we can figure out a way to refactor and simply a bit, so that journal flushing semantics become more clear.

Reply:
have u tried? just unit test see if flush will be called if only one inode with delete-alluxio-only is being processed?

Reply:
actually if u can see for delete alluxio-only there's at least any batch flush happening, then that's fine, means this fix serves its purpose.